File: app/src/main/java/by/egorgutko/autorization/data/RoomDB/DateBase.java
package by.egorgutko.autorization.data.RoomDB;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Single;

public class DateBase implements DBInterface  {

    private TaskDao taskDao;

    public DateBase(TaskDao taskDao) {
        this.taskDao = taskDao;
    }

    @Override
    public Single<List<PersonTasks>> getTaskList(String name) {
        return Single.fromCallable(()-> taskDao.getTaskList(name));
    }

    @Override
    public Completable setUserTask(PersonTasks task) {
        return Completable.fromAction(()-> taskDao.setTaskOfUSer(task));
    }
}

File: app/src/main/java/by/egorgutko/autorization/data/RoomDB/DateBase.kt
package by.egorgutko.autorization.data.RoomDB

import io.reactivex.Completable
import io.reactivex.Single

class DateBase(private val taskDao: TaskDao) : DBInterface {

    override fun getTaskList(name: String): Single<List<String>> {
        return Single.fromCallable { taskDao.getPersons(name) }
    }

    override fun setUserTask(task: Person): Completable {
        return Completable.fromAction { taskDao.setTaskOfUSer(task) }
    }
}
